src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -794,21 +786,46 @@ private static void CheckLambdaConversion(LambdaSymbol lambdaSymbol, TypeSymbol
                 var lambdaParameter = lambdaSymbol.Parameters[p];
                 var delegateParameter = delegateParameters[p];
 
-                if (lambdaParameter.HasExplicitDefaultValue &&
-                    lambdaParameter.ExplicitDefaultConstantValue is { IsBad: false } lambdaParamDefault)
+                // If synthesizing a delegate with `decimal`/`DateTime` default value,
```

Comment:
>// If synthesizing a delegate with `decimal`/`DateTime` default value, [](http://example.com/codeflow?start=16&length=70)

How do we know that this is the case?

Reply:
Specifically the part about "synthesizing a delegate"

Reply:
Good catch, thanks. Will add a check.